Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/CommonComponents/LocationDrawingsButton.tsx

```diff
@@ -0,0 +1,160 @@
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  01/06/2024 - Collins Self
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import React from 'react';
+import { BtnDropdown, GenericController, LoadingScreen, ServerErrorIcon, ToolTip } from '@gpa-gemstone/react-interactive';
+import LocationDrawingsModal from './LocationDrawingsModal';
+import { OpenXDA } from '@gpa-gemstone/application-typings';
+import { CrossMark } from '@gpa-gemstone/gpa-symbols';
+
+interface LocationDrawingsButtonProps {
+    Locations: OpenXDA.Types.Location[];
+}
+
+type DropDownOption = {
+    Label: JSX.Element | string,
+    Callback: () => void,
+    Disabled: boolean
+    ToolTipContent: JSX.Element,
+    ShowToolTip: boolean,
+    ToolTipLocation: ('top' | 'bottom' | 'left' | 'right'),
+    Key: string | number
+}
+
+const LocationDrawingsButton: React.FC<LocationDrawingsButtonProps> = (props) => {
+    const [hover, setHover] = React.useState<'none' | 'drawings'>('none');
+    const [pageState, setPageState] = React.useState<"loading" | "error" | "idle">("idle");
+    const [selectedLocation, setSelectedLocation] = React.useState<OpenXDA.Types.Location>();
+    const multipleLocations = React.useMemo(() => props.Locations.length > 1, [props.Locations]);
+    const [showDrawingsModal, setShowDrawingsModal] = React.useState<boolean>(false);
+    const [locationOptions, setLocationOptions] = React.useState<DropDownOption[]>([]);
+    const LocationDrawingController = new GenericController(`${homePath}api/LocationDrawing`, "Name", true);
+
+    const isValid = (location: OpenXDA.Types.Location, drawingData) => {
+        let e = "";
+
+        if (!location
+            || (location.Alias == ""
+                && location.Description == ""
+                && location.ID == 0
+                && location.Latitude == null
+                && location.LocationKey == ""
+                && location.Longitude == null
+                && location.Name == ""))
```

Comment:
```suggestion
            || (location.Alias === ""
                && location.Description === ""
                && location.ID === 0
                && location.Latitude == null
                && location.LocationKey === ""
                && location.Longitude == null
                && location.Name === ""))
```

We should use strict equality when we do comparisons to non-nullish values.